.drone.yml

```diff
@@ -0,0 +1,437 @@
+---
+kind: pipeline
+type: kubernetes
+name: lint-files
+
+platform:
+  os: linux
+  arch: amd64
+
+trigger:
+  event:
+    exclude:
+      - promote
+
+steps:
+  - name: lint_dockerfiles
+    pull: if-not-exists
+    image: hadolint/hadolint:latest-alpine
+    commands:
+      - hadolint Dockerfile
+
+  - name: lint_helm_files
+    image: pelotech/drone-helm3
+    settings:
+      mode: lint
+      chart: ./helm
+      values_files:
+        - ./helm/values/dev-values.yaml
+        - ./helm/values/test-values.yaml
+        - ./helm/values/prod-values.yaml
+
+
+---
+kind: pipeline
+type: kubernetes
+name: promotion-check
+
+platform:
+  os: linux
+  arch: amd64
+
+trigger:
+  event:
+    - promote
+
+steps:
+  - name: branch-check
+    pull: if-not-exists
+    image: alpine:latest
+    commands:
+      - echo Cannot promote from non 'main' branch
+      - exit 1
+    when:
+      branch:
+        exclude:
+          - main
+
+  - name: environment-check
+    pull: if-not-exists
+    image: alpine:latest
+    commands:
+      - echo Cannot promote to a non prod environment
+      - exit 1
+    when:
+      target:
+        exclude:
+          - production
+
+  - name: build-check
+    pull: if-not-exists
+    image: drone/cli:1.6.2-alpine
+    environment:
+      DRONE_BUILD_PARENT: ${DRONE_BUILD_PARENT}
+      DRONE_SERVER: https://drone-gh.acp.homeoffice.gov.uk
+      DRONE_TOKEN:
+        from_secret: drone_token
+    commands:
+      - chmod +x ./pipeline-scripts/check-build-promotion.sh
```

Comment:
I think promotions will fail as it doesn't look like this script has been copied in to this repo.
Another alternative would be to place the script somewhere central and curl it. I think the UI does it. A bit more work though.

Reply:
Good spot. Added .
Also added a tech Huddle Mural ticket to look into ways to centralise this type of scripts

Reply:
Good idea!